visallo/visallo
# Sanitize filenames before writing them to the local filesystem

When a user drops a file onto Visallo, the web server writes it to a temporary file on its local storage. Depending on the filesystem in use, certain characters are not allowed. This PR changes the code to use a generated temp file name rather than trying to sanitize and use the original filename. As part of this, the system will also set the original filename as the property on the vertex rather than the processed name.


Testing Instructions:
1. Upload files with illegal characters and percent encoding sequences.
2. Ensure that the upload succeeds and that the vertex displays the original filename

Points of Regression:
None

CHANGELOG
Fixed: Uploading files with illegal characters in their names
Added: The temporary directory that file uploads are written to can be configured via the property `org.visallo.web.routes.vertex.VertexImport.tempDir`
